### PR TITLE
feat: implement `Emoji` object

### PR DESCRIPTION
**Pull request by Dworv**
This PR implements the `Emoji` object into the library.

### CODE REVIEW

**Comment by Dworv** [comment]
#14

File: retux/client/resources/emoji.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from attrs import define, field
+from .abc import Object, Snowflake
+from .user import User
+from ...utils import optional_c
+
+@define(kwargs_only=False)
```
Comment (i0bs) [inline retux/client/resources/emoji.py:6]:
```suggestion
@define()
```

File: retux/client/resources/emoji.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+class Emoji(Object):
```
Comment (i0bs) [inline retux/client/resources/emoji.py:7]: What's the reasoning behind making it an object subclass?

File: retux/client/resources/emoji.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    """
+    Represents an Emoji from Discord.
+    
+    Attributes
+    ----------
+    id : `Snowflake`, optional
+        The ID of the Emoji.
+
+        Only availible on guild created emojis.
```
Comment (i0bs) [inline retux/client/resources/emoji.py:16]:
```suggestion
        Only available on guild created emojis.
```

File: retux/client/resources/emoji.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        Whether or not the emoji is availible.
+
+        This may be false if the server that owns 
+        the emoji lost a level of boosting.
+    """
+    id: str | Snowflake | None = field(converter=optional_c(Snowflake), default=None)
+    """
+    The ID of the Emoji.
+    
+    Only availible on guild created emojis.
```
Comment (i0bs) [inline retux/client/resources/emoji.py:43]:
```suggestion
    Only available on guild created emojis.
```

File: retux/client/resources/emoji.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        The ID of the Emoji.
+
+        Only availible on guild created emojis.
+    name : `str`, optional
+        The name of the emoji.
+
+        This is a unicode emoji if the emoji 
+        is standard, otherwise it is the name 
+        of the emoji.
+    roles : `list[Role]`, optional
```
Comment (i0bs) [inline retux/client/resources/emoji.py:23]:
```suggestion
```

File: retux/client/resources/emoji.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        A list of roles allowed to use this emoji.
```
Comment (i0bs) [inline retux/client/resources/emoji.py:24]:
```suggestion
```

File: retux/client/resources/emoji.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    user : `User`, optional
+        The user that created the emoji.
+    require_colons : `bool`, optional
+        Whether or not the emoji needs to be wrapped in colons.
+    managed : `bool`, optional
+        Whether or not the emoji is managed.
+    animated : `bool`, optional
+        Whether or not the emoji is animated.
+    availible : `bool`, optional
```
Comment (i0bs) [inline retux/client/resources/emoji.py:33]:
```suggestion
    available : `bool`
```

File: retux/client/resources/emoji.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        Whether or not the emoji is availible.
```
Comment (i0bs) [inline retux/client/resources/emoji.py:34]:
```suggestion
        Whether or not the emoji is available.
```

File: retux/client/resources/emoji.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+        This may be false if the server that owns
```
Comment (i0bs) [inline retux/client/resources/emoji.py:36]:
```suggestion
        Defaults to `False` if the server that owns 
```

File: retux/client/resources/emoji.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    # """A list of roles allowed to use this emoji."""
+    user: dict | User | None = field(converter=optional_c(User))
+    """The user that created the emoji."""
+    require_colons: bool | None = field(default=None)
+    """Whether or not the emoji needs to be wrapped in colons."""
+    managed: bool | None = field(default=None)
+    """Whether or not the emoji is managed."""
+    animated: bool | None = field(default=None)
+    """Whether or not the emoji is animated."""
+    availible: bool | None = field(default=None)
```
Comment (i0bs) [inline retux/client/resources/emoji.py:64]:
```suggestion
    available: bool = field(default=False)
```

File: retux/client/resources/emoji.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    """
+    Whether or not the emoji is availible.
```
Comment (i0bs) [inline retux/client/resources/emoji.py:66]:
```suggestion
    Whether or not the emoji is available.
```

File: retux/client/resources/emoji.py [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    
+    This may be false if the server that owns
```
Comment (i0bs) [inline retux/client/resources/emoji.py:68]:
```suggestion
    Defaults to `False` if the server that owns 
```

**Review by i0bs — CHANGES_REQUESTED** [review]

File: retux/client/resources/emoji.py (hunk 2 above)
Comment (Dworv) [reply retux/client/resources/emoji.py:7]: I thought we were meant to subclass `Object` for classes with IDs

**Review by i0bs — APPROVED** [review]
LGTM, thank you for contributing to retux!

**Pull request merged**